.CMake/apple.cmake

```diff
@@ -0,0 +1,1101 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
This would change the nature of the project as a "clean" MIT-licensed project.

Reply:
The script enables versioned and separated output for easy consumption by the consumers who prefer  a single command to build all platform outputs at one time. However, it's of lesser value to generic users as it is opinionated.

From a tests perspective, unsure how those would be implemented due to the script *requiring* the use of openssl. Particularly, openssl output in the same versioned structure. Some openssl build projects use similar variants to this structure but which openssl build project is used depends on the consumer's needs. In my case, I leverage openssl_for_android for wrapping droid builds and OpenSSL for iOS / macOS builds.

The only real required changes are for the compile mods listed in the PR. I had hopes the script would be useful to others but it's not required.

Reply:
> I had hopes the script would be useful to others but it's not required

I think it does -- but I'm unsure whether this is the right place. What about giving it a home in [oqs-demos](https://github.com/open-quantum-safe/oqs-demos/tree/main/openssl3) next to the Docker image, promising to deliver a cross-platform binary? This then basically serves the same purpose as the Dockerfile, namely to alleviate users of the need to have to pick-and-parcel components to get a runnable image.

> on second thought the script is too specific to be of general use

This I'd second.

Reply:
As we agree this is too specific to be of general use (and as it doesn't have a dedicated CI run to ensure it "stays good" _and_ has code with different licensing terms) I don't think this script should be within `oqs-provider`. That said, I do think it has value, but may better be placed within [oqs-demos](https://github.com/open-quantum-safe/oqs-demos). Tagging @SWilson4 @praveksharma for their opinion. I'd think the same argumentation goes for the [related PR in liboqs](https://github.com/open-quantum-safe/liboqs/pull/1717).

Reply:
> As we agree this is too specific to be of general use (and as it doesn't have a dedicated CI run to ensure it "stays good" _and_ has code with different licensing terms) I don't think this script should be within `oqs-provider`. That said, I do think it has value, but may better be placed within [oqs-demos](https://github.com/open-quantum-safe/oqs-demos). Tagging @SWilson4 @praveksharma for their opinion. I'd think the same argumentation goes for the [related PR in liboqs](https://github.com/open-quantum-safe/liboqs/pull/1717).

I agree that oqs-demos location would a better home. That said, I don't think I have bandwidth to offer support on or maintain this script, and I would hesitate to add another component into a subproject for which we already have tenuous support issues (i.e., red CI) without some sort of maintenance commitment (probably from @andybrucenet).

Reply:
@andybrucenet Would/Could you commit to maintaining this script when it'd land in `oqs-demos`? Either way, I'd suggest to close it here requesting a PR at `oqs-demos` if you'd be willing to provide a test for it there as well as maintain it. The same thought applies to https://github.com/open-quantum-safe/liboqs/pull/1717.